x-pack/solutions/security/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/upgrade_with_preview.md

```diff
@@ -0,0 +1,706 @@
+
+- **Customized field**: a prebuilt rule's field that has a value that differs from the original field value of the originally installed prebuilt rule.
+
+- **Non-customized rule**: a prebuilt rule that doesn't have any customized fields.
+
+- **Customized rule**: a prebuilt rule that has one or more customized fields.
+
+### Assumptions
+
+- Below scenarios only apply to prebuilt detection rules.
+- A prebuilt rule is shown in the Rule Upgrade table when there's a newer version of this rule in the currently installed package with prebuilt rules.
+
+### Functional requirements
+
+- User should be able to upgrade prebuilt rules one-by-one with the ability to preview:
+  - what updates they would receive from Elastic in the latest rule version;
+  - what user customizations they would retain in the rule;
+  - are there any conflicts between the updates from Elastic and the user customizations;
+  - what they should pay attention to in case there are any conflicts.
+- A preview should be shown in the Rule Upgrade flyout.
+- The Rule Upgrade flyout should contain a few tabs:
+  - The "Updates" tab.
+  - The "Overview" tab.
+  - The "Investigation guide" tab.
+- On the "Updates" tab:
+  - We should show the updates from Elastic and the user-customized fields.
+  - We should show only those fields that are [customizable](./shared_assets/customizable_rule_fields.md).
+  - We shouldn't show technical fields and those that are [not customizable](./shared_assets/non_customizable_rule_fields.md).
+- User should be able to upgrade a prebuilt rule that has some updates to [non-customizable fields](./shared_assets/non_customizable_rule_fields.md) in the latest version.
+- Any other fields that are not involved in the rule upgrade workflow, such as `enabled` or `exceptions`, should stay unchanged after rule upgrade.
+
+## Scenarios
+
+### Rule upgrade field preview
+
+#### Preview non-customized field that has an upgrade (AAB)
+
+**Automation**: Jest functional test for each \<field\>.
+
+```Gherkin
+Given an installed prebuilt rule
+And that rule has no customizations
+And <field> has an upgrade
+When user opens the Rule Update Flyout
+Then user should see <field> has no conflicts
+And <field> is shown collapsed
+When user expands <field>
+Then user should see <field> Diff View
+And user should see <field> Readonly View
+
+Examples:
+<field> = all customizable fields
+```
+
+#### Preview customized field that doesn't have an upgrade (ABA)
+
+**Automation**: Jest functional test for each \<field\>.
+
+```Gherkin
+Given an installed prebuilt rule
+And <field> is customized
+And <field> has no upgrades
+When user opens the Rule Update Flyout
+Then user should see <field> has a customized value
+And <field> has no conflicts
+And <field> is shown collapsed
+When user expands <field>
+Then user should see <field> Diff View
+And user should see <field> Readonly View
+
+Examples:
+<field> = all customizable fields
```

Comment:
Similarly, in this and other tests from this section we could elaborate more on the state of the flyout.